script_library/run-ansible.sh

```diff
@@ -29,21 +33,18 @@ function run_ansible(){
     fi
 
     if [[ -f ${extravarsfile} ]]; then
-        echo "Extra variables file exists: $(realpath ${extravarsfile}). Loading its vars in ansible-playbook call."
-        extra_vars="-e @${extravarsfile}"
+        echo "Extra variables file exists: $(realpath "${extravarsfile}"). Loading its vars in ansible-playbook call."
+        ansible_command="$ansible_command -e @${extravarsfile}"
     fi
 
-    if [[ -f ${inventorydir} ]]; then
-        echo "Inventory directory (${inventorydir}) exists, adding it to the ansible-playbook call."
-        inventory="-i ${inventorydir}"
-    fi
     if [[ ${USE_ARA:-False} == "True" ]]; then
         echo "Loading ARA"
-        source ${HOME}/.socok8svenv/ara.rc
+        # shellcheck disable=1090
+        source "${HOME}"/.socok8svenv/ara.rc
     fi
 
-    pushd ${socok8s_absolute_dir}
-        ansible-playbook ${extra_vars:-} -i ${inventorydir} $@ -v
-    popd
+    pushd "${socok8s_absolute_dir}" || exit
+        ${ansible_command} -i "${inventorydir}" "$@" -v
```

Comment:
are we sure this is passing arguments properly, if multiple arguments in $@ are provided? I would have the impression they would be received as one, but don't trust my bash foo.